csrc/dynamic_transform.cpp

```diff
@@ -457,8 +457,22 @@ void DynamicTransformConcretizer::concretize() {
 
   // Finally, propagate concretized domains
   auto all_stmts = StmtSort::getStmts(info_->fusion());
+  std::unordered_set<TensorView*> visited_tvs;
   for (auto tv : ir_utils::filterByType<TensorView>(all_stmts)) {
-    mutate(tv);
+    if (tv->definition()) {
+      for (auto outp :
+           ir_utils::filterByType<TensorView>(tv->definition()->outputs())) {
+        if (visited_tvs.find(outp) == visited_tvs.end()) {
+          mutate(outp);
+          visited_tvs.insert(tv);
+        }
+      }
+    } else {
+      if (visited_tvs.find(tv) == visited_tvs.end()) {
+        mutate(tv);
+        visited_tvs.insert(tv);
+      }
+    }
```

Comment:
This is awkward and inefficient, and this functionally should probably be implemented as an option to `StmtSort::getStmts` instead.

Reply:
Pushed an alternative that adds a `traverse_siblings` option for `IterVisitor` and `StmtSort`.